### PR TITLE
Filter Program Offers to only show offers for the current site

### PR DESCRIPTION
**Pull request by bill-filler**
Program Offers are now site-aware and the view is filtered to only show
offers for the current site. Site field also added to dashboard/offers
UI when creating/displaying offers from the dashboard.

WL-1152

### CODE REVIEW

**Comment by bill-filler** [comment]
I think the codecov is failing because of the oscar templates (which are large html files) I had to include to add Site to the summary and details page:
https://github.com/edx/ecommerce/pull/1456/files#diff-bd70d183ff3ef2312b933085571eb273
https://github.com/edx/ecommerce/pull/1456/files#diff-89356728367e81c60819b61474c73c72
Not sure what to do about that.

**Comment by mjfrey** [comment]
I would add a view test that submits a form so the new code that stores the site in the kwargs gets tested.

File: ecommerce/extensions/dashboard/offers/forms.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from oscar.apps.dashboard.offers.forms import MetaDataForm as CoreMetaDataForm
+from oscar.core.loading import get_model
+
+
+class MetaDataForm(CoreMetaDataForm):
+    class Meta:  # pylint: disable=C1001
```
Comment (clintonb) [inline ecommerce/extensions/dashboard/offers/forms.py]: Don't disable this. Fix it: `class Meta(object)`.

File: ecommerce/programs/views.py [hunk 2]
```diff
@@ -24,7 +24,9 @@ def get_context_data(self, **kwargs):
         return context
 
     def get_queryset(self):
+        site = self.request.site
```
Comment (clintonb) [inline ecommerce/programs/views.py]: Nit: Merge this with the line below. The `site` variable isn't necessary.

File: ecommerce/templates/oscar/dashboard/offers/offer_detail.html [hunk 3]
```diff
@@ -0,0 +1,102 @@
+{% extends 'dashboard/layout.html' %} {% load currency_filters %} {% load i18n %} {% block title %} {% blocktrans with name=offer.name %} {{ name }} | Offers {% endblocktrans %} | {{ block.super }} {% endblock %} {% block breadcrumbs %}
```
Comment (clintonb) [inline ecommerce/templates/oscar/dashboard/offers/offer_detail.html]: Break this up into multiple lines. Each `extends` and `load` call is on its own line. Each `block` is its own block.

File: ecommerce/templates/oscar/dashboard/offers/summary.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+        <ul>
+            {% for restriction in offer.availability_restrictions %}
+            <li>{{ restriction.description }}</li>
+            {% endfor %}
+        </ul>
+        {% else %} - {% endif %}
+    </div>
+    {% endif %}
+</div>
```
Comment (clintonb) [inline ecommerce/templates/oscar/dashboard/offers/summary.html]: Add a new line.

File: ecommerce/templates/oscar/dashboard/offers/offer_detail.html [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        <tr>
+            <td><a href="{% url 'dashboard:order-detail' number=order.number %}">{{ order.number }}</a></td>
+            <td>{{ order.date_placed }}</td>
+            <td>{{ order.total_incl_tax|currency }}</td>
+            <td>{{ discount.amount|currency }}</td>
+        </tr>
+        {% endwith %} {% endfor %}
+    </tbody>
+</table>
+{% include 'dashboard/partials/pagination.html' %} {% endif %} {% endblock dashboard_content %}
```
Comment (clintonb) [inline ecommerce/templates/oscar/dashboard/offers/offer_detail.html]: Add a new line.

**Review by clintonb — CHANGES_REQUESTED** [review]
You need tests. Codecov doesn't look at HTML templates. You've added Python code that is not tested.

**Review by clintonb — APPROVED** [review]
Remember to squash commits.

**Pull request merged**